### DISCUSSION TITLE
Provide "File Storage" feature to users

### DISCUSSION THREAD

**Discussion by StachuDotNet** [darklang/dark#4316] [Feedback]
The intention behind File Storage is to house and serve user files. (e.g. documents, media, etc.)

This is separate from the existing "Static Assets" feature, which is aimed at housing and serving static assets of a front-end.

**Comment by pbiggar** [comment]
One of the thing that's unique about this is that users will often want to upload files indirectly. That is, instead of uploading a file to the server, the server will provide a URL which can be used to upload the file directly to Cloud Storage.
